### PR TITLE
[next-auth] Update signin with third parameter

### PR DESCRIPTION
**Pull request by luizwhite**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [signIn additional params](https://next-auth.js.org/getting-started/client#additional-params)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by luizwhite** [comment]
Kind of new to this, so why local type test `npm test next-auth` complains about react, but not the CI tests?
When I install the @types/react locally all is fine though

image

File: types/next-auth/client.d.ts [hunk 1]
```diff
@@ -52,6 +52,7 @@ declare function signin(
         callbackUrl?: string;
         redirect?: boolean;
     },
+    authorizationParams?: GenericObject
```
Comment (lluia) [inline types/next-auth/client.d.ts:55]: [In the source code](https://github.com/nextauthjs/next-auth/blob/main/src/client/index.js#L232) I see this ends passed to the `URLSearchParams()` constructor. What this constructor expects is actually a more mixed type than just an object; I think we should then update to 📐
```ts
authorizationParams?: string | string[][] | GenericObject | URLSearchParams
```

**Review by 0ubbe — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution 💯

I small suggestion 💬  to try to refine more the type of `authorizationParams` 👍🏼

File: types/next-auth/client.d.ts (hunk 1 above)
Comment (luizwhite) [reply types/next-auth/client.d.ts:55]:
```suggestion
    authorizationParams?: string | string[][] | GenericObject | URLSearchParams
```

Ohh, i didn't see that, thx!
So, we better add that then

**Review by 0ubbe — APPROVED** [review]
Thanks 💚 💯

**Comment by luizwhite** [comment]
Ready to merge

**Pull request merged**